helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -837,7 +835,7 @@ private static void addCompletedTasks(Set<Integer> set, JobContext ctx, Iterable
    * @param state
    * @return
    */
-  private boolean isTaskNotInTerminalState(TaskPartitionState state) {
+  protected static boolean isTaskNotInTerminalState(TaskPartitionState state) {
```

Comment:
You might want to move it to TaskPartitionState class.

Reply:
TaskPartitionState is just an enum that lists all the states for the tasks. So I think it is better to keep it here like other similar methods.